Canvas/Canvas/CanvasAppDelegate.swift

```diff
@@ -360,3 +360,41 @@ extension AppDelegate: NativeLoginManagerDelegate {
         }, completion:nil)
     }
 }
+
+extension AppDelegate: BackgroundURLSessionDelegateEventHandler {
+    func urlSessionDidFinishEvent(forBackgroundURLSession session: URLSession) {
+        environment.logger.log(#function)
+        if let userID = TheKeymaster.currentClient?.currentUser?.id {
+            let createSubmissions = CreateFileSubmissions(env: environment, userID: userID)
+            environment.queue.addOperation(createSubmissions)
+        }
+    }
+
+    func urlSessionDidFinishEvents(forBackgroundURLSession session: URLSession) {
+        environment.logger.log("app delegate \(#function)")
+        let operation = OperationSet()
+        for client in FXKeychain.shared().clients() {
+            guard let entry = client.keychainEntry else {
+                continue
+            }
+            let env = AppEnvironment()
+            env.userDidLogin(session: entry)
+            let createSubmissions = CreateFileSubmissions(env: env, userID: entry.userID)
+            operation.addOperation(createSubmissions)
+        }
+        environment.queue.addOperations([operation], waitUntilFinished: true)
+        environment.backgroundAPIManager.complete(session: session)
+    }
+}
```

Comment:
This looks like a merge error. `BackgroundURLSessionDelegateEventHandler` no longer exists. You can remove this extension. Sorry I didn't notice this before.